src/Game.java

```diff
@@ -0,0 +1,116 @@
+import java.util.HashMap;
+import java.util.Map;
+
+public class Game {
+    private final Board board;
+    private final Map<Integer,Integer> snakeMap;
+    private final Map<Integer,Integer> ladderMap;
+    private boolean gameWon;
+
+    public Game(Board board) {
+        this.board = board;
+        this.snakeMap = new HashMap<>();
+        this.ladderMap = new HashMap<>();
+        this.gameWon = false;
+    }
+
+    public Board getBoard() {
+        return this.board;
+    }
+
+    public boolean getGameWon() {
+        return this.gameWon;
+    }
+
+    public void setGameWon() {
+        this.gameWon = true;
+    }
+
+    public void addSnakeToBoard(int startCoordinate, int endCoordinate) {
+        this.snakeMap.put(startCoordinate, endCoordinate);
+    }
+
+    public void addLadderToBoard(int startCoordinate, int endCoordinate) {
+        this.ladderMap.put(startCoordinate, endCoordinate);
+    }
+
+    /*public boolean isGameWon() {
+        for(Player player : this.board.getPlayerList()) {
+            if(player.getHasWon())
+                return true;
+        }
+        return false;
+    }*/
+
+    private void printMove(Player player, int diceNumber, int start, int end) {
+        System.out.println(player.getPlayerName() + " rolled a " + diceNumber + " and moved from " +
+                           start + " to " + end);
+        return;
+    }
+
+    private int updateDieNumber() {
+        int val1 = 6;
+        int val2 = (int) ((Math.random()*6) + 1);
+        val1 += val2;
+        if(val2 == 6) {
+            int val3 = (int) ((Math.random()*6) + 1);
+            if(val3 == 6)
+                return 0;
+            val1 += val3;
+        }
+        return val1;
+    }
+
+    private int getDieNumber(int numOfDie) {
+        int sum = 0,i;
+        for(i=0;i<numOfDie;i++) {
+            int val = (int) ((Math.random()*6) + 1);
+            if(val == 6)
+                val = updateDieNumber();
+            sum += val;
+        }
+        return sum;
+    }
+
+    private int updateCoordinateBySnakesAndLadders(Tile tile, Player player, int newCoordinate) {
```

Comment:
not clear what it does based on method name

Reply:
Also, it is a get operation and not an update operation that this method does

Reply:
The idea is to advance based on the dice output, and then check if the final new coordinate has any snakes and ladders. Also checks for consecutive snakes and ladders.

Reply:
Yeah I checked the code but the method name should be able to explain that to me.

Reply:
Right, I agree.